Automattic/wp-calypso
# SectionNav: Modifying README to mention the new hasHorizontalScroll prop, and adding tests

Follow-up to https://github.com/Automattic/wp-calypso/pull/89618 and https://github.com/Automattic/wp-calypso/pull/89784

## Proposed Changes

* This PR modifies the `SectionNav` readme to make sure information about the `NavTabs` `hasHorizontalScroll` prop is added.
* It also adds a couple of tests whose purpose is to ensure that setting `hasHorizontalScroll` to true adds the `has-horizontal-scroll` class if a screen width is greater than the mobile version limit defined at 480px, and doesn't if less than that. New: One more test checks that setting `hasHorizontalScroll` to false also doesn't change expected behaviour (for large screens, it doesn't add the new class).

For future improvement:
* Ideally we would test that setting `hasHorizontalScroll` to true also doesn't render the dropdown itself unless on mobile screens. This requires a little more investigation as` requestAnimationFrame` in the `afterLayoutFlush` function doesn't run in the tests as is.

## Testing Instructions

* To test the README, check that the updated instructions make sense.
* To test the SectionNav tests, run `yarn run test-client client/components/section-nav/test`. 

## Pre-merge Checklist
